kong/error.go

```diff
@@ -10,12 +10,14 @@ import (
 type APIError struct {
 	httpCode int
 	message  string
+	raw      []byte
 }
 
-func NewAPIError(code int, msg string) *APIError {
+func NewAPIError(code int, msg string, raw []byte) *APIError {
```

Comment:
How about we don't change this but include a second "constructor" called `NewAPIErrorWithBody` or `NewAPIErrorWithRaw(Body)` and use that in `NewAPIError`?

This way we'll avoid passing the third param `[]byte{}` where it's not needed.

Reply:
👍 + we do not break the API.